compiler/AST/primitive.cpp

```diff
@@ -702,11 +702,12 @@ initPrimitive() {
   prim_def(PRIM_REQUIRE, "require", returnInfoVoid, false, false);
 }
 
-Map<const char*, VarSymbol*> memDescsMap;
+static Map<const char*, VarSymbol*> memDescsMap;
+static Map<int, VarSymbol*> memDescsNodeMap;
```

Comment:
I think it's more likely that somebody will be "suprised" by this use of AST ids than it is that they'll be surprised that we can't delete TypeSymbols. If I am correct that TypeSymbols are never dead-code eliminated, this code would be less unusual if it just used TypeSymbol* as the key to the map. (Or a Type* for that matter).